rbac/management/group/definer.py

```diff
@@ -192,8 +193,9 @@ def remove_roles(group, roles_or_role_ids, tenant, user=None):
                 policy.roles.remove(role)
                 logger.info(f"Removing role {role} from group {group.name} for tenant {tenant.org_id}.")
 
-                dual_write_handler = RelationApiDualWriteGroupHandler(group, ReplicationEventType.UNASSIGN_ROLE, [])
-                dual_write_handler.replicate_removed_role(role)
+                if group.tenant.tenant_name != "public":
```

Comment:
this check is inside  `replicate_removed_role`

Reply:
Yes. It needs to be because the public tenant doesn't have a default workspace. Is there an issue?

Reply:
Hrm yeah the issue is this part in the constructor:

```
            self.tenant = group.tenant
            self.default_workspace = Workspace.objects.get(tenant=self.tenant, type=Workspace.Types.DEFAULT)
```

Thanks for explaining your concern. I think you're right that there is redundancy. The redundancy is probably more inside `replicate_removed_role` and other methods that check the group's tenant. I'll remove those checks and add a check inside the constructor since it wouldn't work in those cases as the design evolved.

Reply:
oh I see, thanks for explaining and I definitely agree 👍